src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -470,6 +503,10 @@ def build_container_from_source_with_buildpack(self, image_name, source):  # pyl
             logger.debug(f"Determined the run image to use as {buildpack_run_image}.")
             command.extend(['--run-image', buildpack_run_image])
 
+        # If the user specifies a target port, pass it to the buildpack
+        if self.target_port:
+            command.extend(['--env', f"PORT={self.target_port}"])
```

Comment:
Just to validate: from your comment in the description of the PR it sound like the PORT env variable value is basically intercepted by our logic in the Buildpack and then used by us to set the port to expose on the final image. Is this correct?

Reply:
That's correct -- the `--env` argument will propagate the given environment variables to the stack that the detect/build phases occur, so our .NET 7 buildpack will have access to this `PORT` value set